packages/idempotency/tests/e2e/makeFunctionIdempotent.test.ts

```diff
@@ -0,0 +1,101 @@
+/**
+ * Test makeFunctionIdempotent
+ *
+ * @group e2e/idempotency
+ */
+import { generateUniqueName, invokeFunction, isValidRuntimeKey } from '../../../commons/tests/utils/e2eUtils';
+import { RESOURCE_NAME_PREFIX, SETUP_TIMEOUT, TEARDOWN_TIMEOUT, TEST_CASE_TIMEOUT } from './constants';
+import { v4 } from 'uuid';
+import { App, Stack } from 'aws-cdk-lib';
+import { DynamoDBClient } from '@aws-sdk/client-dynamodb';
+import { createHash } from 'node:crypto';
+import { deployStack, destroyStack } from '../../../commons/tests/utils/cdk-cli';
+import { GetCommand, ScanCommand } from '@aws-sdk/lib-dynamodb';
+import { createIdempotencyResources } from '../helpers/idempotencyUtils';
+
+const runtime: string = process.env.RUNTIME || 'nodejs18x';
+
+if (!isValidRuntimeKey(runtime)) {
+  throw new Error(`Invalid runtime key value: ${runtime}`);
+}
+const stackName = generateUniqueName(RESOURCE_NAME_PREFIX, v4(), runtime, 'makeFnIdempotent');
```

Comment:
The `uuid` should be the same across the same test, so that we can identify resources related to the same test run by the prefix.

The way it is now each resource name will have its own prefix because we generate a new uuid for each name.

Reply:
oooh that makes sense, will adjust